airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -71,34 +83,68 @@ protected ChangeEvent<String, String> computeNext() {
     while (!MoreBooleans.isTruthy(publisherStatusSupplier.get()) || !queue.isEmpty()) {
       final ChangeEvent<String, String> next;
       try {
-        final Duration waitTime = receivedFirstRecord ? SUBSEQUENT_RECORD_WAIT_TIME : firstRecordWaitTime;
+        // #18987: waitTime here is largely meaningless with heartbeats but leaving for backward
+        // compatibility
+        // for connectors not implementing heartbeat yet (MySql, MSSql)
+        final Duration waitTime = receivedFirstRecord ? SUBSEQUENT_RECORD_WAIT_TIME : this.firstRecordWaitTime;
         next = queue.poll(waitTime.getSeconds(), TimeUnit.SECONDS);
       } catch (final InterruptedException e) {
         throw new RuntimeException(e);
       }
 
       // if within the timeout, the consumer could not get a record, it is time to tell the producer to
       // shutdown.
+      // #18987: Noticed in testing that it's possible for DBZ to be stuck "Searching for WAL resume
+      // position"
+      // when no changes exist. In that case queue will pop after timeout with null value for next
       if (next == null) {
-        LOGGER.info("Closing cause next is returned as null");
+        LOGGER.info("Closing: queue returned null event");
         requestClose();
         LOGGER.info("no record found. polling again.");
         continue;
       }
 
+      // check if heartbeat and read hearbeat position
```

Comment:
nit: What would make this method more readable is to: 
- Add interface to ``CdcTargetPosition`` to indicate whether heartbeats are supported
- Only doing the hearbeat code for those sources that support this

Reply:
Added `isHeartbeatSupported()` and also default implementations rather than touching mysql and mysql.
thanks